src/org/rascalmpl/semantics/dynamic/ShellCommand.java

```diff
@@ -167,32 +244,81 @@ public Result<IValue> interpret(IEvaluator<Result<IValue>> __eval) {
 			case Configuration.TRACING_PROPERTY:
 			  __eval.getConfiguration().setTracing(Boolean.parseBoolean(value));
 			  break;
-			}
+			case Configuration.DEBUGGING_PROPERTY:
+			__eval.getConfiguration().setDebugging(Boolean.parseBoolean(value));
+		}
 
-			__eval.updateProperties();
+		__eval.updateProperties();
+	}
 
+	static public class SetOption extends
+			org.rascalmpl.ast.ShellCommand.SetOption {
+
+		public SetOption(ISourceLocation __param1, IConstructor tree, QualifiedName __param2,
+				Expression __param3, OptionalTerminator term) {
+			super(__param1, tree, __param2, __param3, term);
+		}
+
+		@Override
+		public Result<IValue> interpret(IEvaluator<Result<IValue>> __eval) {
+			String name = "rascal." + ((org.rascalmpl.semantics.dynamic.QualifiedName.Default) this.getName()).fullName();
+			String value = this.getExpression().interpret(__eval).getValue().toString();
+
+			setOption(__eval, name, value);
 			return org.rascalmpl.interpreter.result.ResultFactory.nothing();
 		}
 
 	}
 
 	static public class Test extends org.rascalmpl.ast.ShellCommand.Test {
 
-		public Test(ISourceLocation __param1, IConstructor tree) {
-			super(__param1, tree);
+		public Test(ISourceLocation __param1, IConstructor tree, QualifiedName mut, OptionalTerminator term) {
+			super(__param1, tree, mut, term);
 		}
 
 		@Override
 		public Result<IValue> interpret(IEvaluator<Result<IValue>> __eval) {
-			return org.rascalmpl.interpreter.result.ResultFactory.makeResult(TF.boolType(), VF.bool(__eval.runTests(__eval.getMonitor())), __eval);
+			return org.rascalmpl.interpreter.result.ResultFactory.makeResult(TF.boolType(), VF.bool(__eval.runTests(__eval.getMonitor(), getOptName() != null ? Names.fullName(getOptName()) : null)), __eval);
 		}
 	}
 
+	static public class Undeclare extends org.rascalmpl.ast.ShellCommand.Undeclare {
+
+		public Undeclare(ISourceLocation src, IConstructor node, @Nullable QualifiedName optName, OptionalTerminator terminator) {
+			super(src, node, optName, terminator);
+		}
+
+		@Override
+		public Result<IValue> interpret(IEvaluator<Result<IValue>> eval) {
+			if (getOptName() == null) {
+				// then we clear everything!
+				eval.getCurrentModuleEnvironment().reset();
+			}
+			else {
+				var n = getOptName();
+				if (Names.isQualified(n)) {
+					throw new IllegalArgumentException("name " + Names.fullName(n) + " should not be qualified for :undeclare");
+				}
+
+				var simpleName = Names.name(Names.lastName(n));
+
+				var env = eval.getCurrentModuleEnvironment();
+				
+				env.unsetSimpleVariable(simpleName);
+				env.unsetAllFunctions(simpleName);
+				// TODO: remove ADT from store
```

Comment:
is this a TODO we want to pickup before this PR is merged?

Reply:
it requires a vallang release, so I'm hesitant.